pctasks/core/pctasks/core/cosmos/settings.py

```diff
@@ -165,7 +166,7 @@ def get_async_client(self) -> AsyncCosmosClient:
         else:
             self.ensure_valid_connection_info()
             assert self.url
-            credential = self.key or DefaultAzureCredential()
+            credential = self.key or azure.identity.aio.DefaultAzureCredential()
```

Comment:
This was an issue. We were using the (potentially blocking) sync `DefaultAzureCredential` on the async path.